### PR TITLE
UCP/AM: Support for eager/rndv send flags

### PR DESCRIPTION
**Pull request by brminich**
## What
Support for `UCP_AM_SEND_FLAG_EAGER` and  `UCP_AM_SEND_FLAG_RNDV` in `ucp_am_send_nbx`

### CODE REVIEW

**Review by hoopoepg — APPROVED** [review]

**Comment by brminich** [comment]
@yosefe, please review

File: src/ucp/api/ucp.h [hunk 1]
```diff
@@ -2805,6 +2805,8 @@ ucs_status_ptr_t ucp_am_send_nbx(ucp_ep_h ep, unsigned id,
  * @param [in]  count      Number of elements to receive into @a buffer.
  * @param [in]  param      Operation parameters, see @ref ucp_request_param_t.
  *
+ * @return NULL                 - The receive operation was completed
+ *                                immediately.
```
Comment (yosefe) [inline src/ucp/api/ucp.h:2809]: do we need to change for TAG_RECV_NBX as well?
let's have all doc changes in separate PR

File: src/ucp/core/ucp_am.c [hunk 2]
```diff
@@ -771,16 +771,38 @@ static void ucp_am_send_req_init(ucp_request_t *req, ucp_ep_h ep,
...
+    if (flags & UCP_AM_SEND_FLAG_EAGER) {
+        return SIZE_MAX;
+    } else if (flags & UCP_AM_SEND_FLAG_RNDV) {
+        *max_short = -1; /* disable short, rndv is explicitly requested */
+        return 0;
+    } else {
+        ucp_request_param_rndv_thresh(req, param, &ep_config->rndv.rma_thresh,
+                                      &ep_config->rndv.am_thresh,
+                                      &rndv_rma_thresh, &rndv_am_thresh);
+
```
Comment (yosefe) [inline src/ucp/core/ucp_am.c:790]: can remove space line

File: test/gtest/ucp/test_ucp_am.cc [hunk 3]
```diff
@@ -578,6 +578,11 @@ UCS_TEST_P(test_ucp_am_nbx_dts, long_zcopy_send, "ZCOPY_THRESH=1",
     test_am(64 * UCS_KBYTE);
 }
 
+UCS_TEST_P(test_ucp_am_nbx_dts, send_eager_flag, "RNDV_THRESH=128")
+{
+    test_am(64 * UCS_KBYTE, UCP_AM_SEND_FLAG_EAGER);
```
Comment (yosefe) [inline test/gtest/ucp/test_ucp_am.cc:583]: can we somehow test it actually used rndv?

File: src/ucp/api/ucp.h (hunk 1 above)
Comment (brminich) [reply src/ucp/api/ucp.h:2809]: ok will remove here

File: src/ucp/core/ucp_am.c (hunk 2 above)
Comment (brminich) [reply src/ucp/core/ucp_am.c:790]: ok

File: test/gtest/ucp/test_ucp_am.cc (hunk 3 above)
Comment (brminich) [reply test/gtest/ucp/test_ucp_am.cc:583]: it uses eager and it does this check in `am_data_handler` on line 426
`EXPECT_FALSE(rx_param->recv_attr & UCP_AM_RECV_ATTR_FLAG_RNDV);`

Comment (yosefe) [reply test/gtest/ucp/test_ucp_am.cc:583]: how about expecting flag RNDV in the FLAG_RNDV case?

Comment (brminich) [reply test/gtest/ucp/test_ucp_am.cc:583]: it is done on line 603

**Review by yosefe — APPROVED** [review]

**Comment by brminich** [comment]
@yosefe, all tests passed, except io-demo. Need to rerun?

**Comment by brminich** [comment]
nm, tests will be restarted after squash

**Comment by brminich** [comment]
@yosefe, all tests passed, except iodemo, which failed 3 times in a row with
```
##[error]Failed to connect to remote machine. Verify the SSH service connection details. Error: Error: Timed out while waiting for handshake.
 ```

**Pull request merged**